virtio-devices/src/balloon.rs

```diff
@@ -94,6 +95,59 @@ pub struct VirtioBalloonConfig {
     actual: u32,
 }
 
+#[derive(Clone, Debug)]
+struct PartiallyBalloonedPage {
+    addr: u64,
+    bitmap: Vec<u64>,
+}
+
+impl PartiallyBalloonedPage {
+    fn new() -> Self {
+        let size = get_page_size();
+        let len = ((size >> VIRTIO_BALLOON_PFN_SHIFT) + 63) / 64;
+        Self {
+            addr: 0,
+            bitmap: vec![0; len as usize],
+        }
+    }
+
+    fn pfn_match(&self, addr: u64) -> bool {
+        let page_size: u64 = get_page_size();
+        self.addr == addr & !(page_size - 1)
+    }
+
+    fn bitmap_full(&mut self, page_size: u64) -> bool {
```

Comment:
When the padding bits are pre-initialized as `1`s, I think this function can be simplified a lot, something like: 
```
    fn bitmap_full(&mut self) -> bool {
        self.bitmap.iter().all(|b| *b == u64::MAX)
    }
```

Reply:
Great! It's better enough.